src/components/charts/SavedValues/SavedValuesList.tsx

```diff
@@ -0,0 +1,263 @@
+import { ReactComponent as Edit } from '../../../assets/icons/ic-pencil.svg'
+import { ReactComponent as Delete } from '../../../assets/icons/ic-delete-interactive.svg'
+import { BreadCrumb, ErrorScreenManager, Progressing, showError, useBreadcrumb } from '../../common'
+import { SavedValueListResponse, SavedValueType } from './types'
+import EmptyState from '../../EmptyState/EmptyState'
+import { deleteChartValues, getChartValuesTemplateList } from '../charts.service'
+import './savedValues.scss'
+import PageHeader from '../../common/header/PageHeader'
+import { toast } from 'react-toastify'
+
+export default function SavedValuesList() {
+    const history: RouteComponentProps['history'] = useHistory()
+    const { chartId } = useParams<{ chartId }>()
+    const [loader, setLoader] = useState(false)
+    const [searchApplied, setSearchApplied] = useState(false)
+    const [searchText, setSearchText] = useState('')
+    const [savedValueList, setSavedValueList] = useState<SavedValueType[]>([])
+    const [filteredSavedValueList, setFilteredSavedValueList] = useState<SavedValueType[]>([])
+    const [errorStatusCode, setErrorStatusCode] = useState(0)
+    const [noResults, setNoResults] = useState(false)
+
+    useEffect(() => {
+        getData()
+    }, [])
+
+    const getData = (): void => {
+        setLoader(true)
+        getChartValuesTemplateList(chartId)
+            .then((response: SavedValueListResponse) => {
+                let list = response.result || []
+                list = list
+                    .map((item) => {
+                        return { ...item, isLoading: false }
+                    })
+                    .sort((a, b) => a['name'].localeCompare(b['name']))
+                setSavedValueList(list)
+                setFilteredSavedValueList(list)
+                setLoader(false)
+            })
+            .catch((error) => {
+                setLoader(false)
+                setErrorStatusCode(error.code)
+                showError(error)
+            })
+    }
+
+    const createNewChartValue = (): void => {
+        let link = `${URLS.CHARTS}/discover/chart/${chartId}/saved-values/0`
+        history.push(link)
+    }
+
+    const editChartValue = (chartValueId: number): void => {
+        let link = `${URLS.CHARTS}/discover/chart/${chartId}/saved-values/${chartValueId}`
+        history.push(link)
+    }
+
+    const deleteChartValue = (chartValueId: number): void => {
+        deleteChartValues(chartValueId)
+            .then((response) => {
+                toast.success('Deleted')
+                getData()
+            })
+            .catch((error) => {
+                showError(error)
+            })
+    }
+
+    const handleFilterChanges = (_searchText: string): void => {
+        const _filteredData = savedValueList.filter((cluster) => cluster.name.indexOf(_searchText) >= 0)
+        setFilteredSavedValueList(_filteredData)
+        setNoResults(_filteredData.length === 0)
```

Comment:
We don't need an additional state. We can always use `filteredSavedValueList.length`.

Reply:
There are different images and action in case of no results from DB and no results found after search

Reply:
For no results from DB, we can use `savedValueList.length` whereas for search we can use `filteredSavedValueList.length`, right?

Reply:
Fixed